packages/core-app-api/src/lib/AuthConnector/DefaultAuthConnector.ts

```diff
@@ -77,12 +81,37 @@ export class DefaultAuthConnector<AuthSession>
       provider,
       joinScopes = defaultJoinScopes,
       oauthRequestApi,
+      usePopup,
       sessionTransform = id => id,
     } = options;
 
     this.authRequester = oauthRequestApi.createAuthRequester({
       provider,
-      onAuthRequest: scopes => this.showPopup(scopes),
+      onAuthRequest: async scopes => {
+        if (!usePopup) {
+          // modal before redirect
+          const scope = this.joinScopesFunc(scopes);
+          const redirectUrl = await this.buildUrl('/start', {
+            scope,
+            origin: window.location.origin,
+            redirectUrl: window.location.href,
+            authType: 'redirect',
+          });
+
+          if (provider.hasOwnProperty('provider_id')) {
+            // set the sign in provider here
+            localStorage.setItem(
+              '@backstage/core:SignInPage:provider',
+              provider.provider_id!,
+            );
+          }
+
+          window.location.href = redirectUrl;
+          // we need to return to exit function or else popup occurs
+          return Promise.resolve({} as AuthSession);
+        }
+        return this.showPopup(scopes);
```

Comment:
I think it would be neat create a method for all of the code above with a similar abstraction to the `showPopup` one. Leaving just the minimal logic in this callback:

```ts
if (usePopup) {
  return this.showPopup(scopes)
} else {
  return this.executeRedirect(scopes)
}
```